stats.go

```diff
@@ -107,3 +128,29 @@ func (s *statsExporter) submitMetric(v *view.View, row *view.Row, metricName str
 		return fmt.Errorf("aggregation %T is not supported", v.Aggregation)
 	}
 }
+
+func calculatePercentile(percentile float64, buckets []float64, countPerBucket []int64) float64 {
+	cumulativePerBucket := make([]int64, len(countPerBucket))
+	var sum int64
+	for n, count := range countPerBucket {
+		sum += count
+		cumulativePerBucket[n] = sum
+	}
+	atBin := int64(math.Floor(percentile * float64(sum)))
+
+	var previousCount int64
+	for n, count := range cumulativePerBucket {
+		if atBin >= previousCount && atBin <= count && n < len(buckets) {
+			return buckets[n]
+		}
+		previousCount = count
+	}
+	return buckets[len(buckets)-1]
+}
+
+func buildMetricNameForPercentile(percentile float64) string {
```

Comment:
It might be easier to just call it `percentileName`

Reply:
Fixed.